### ISSUE TITLE
Add-on descriptions registered too early

### ISSUE THREAD

**Issue by Noordfrees** [widelands/widelands#4907]
Currently the EditorGameBase registers the descriptions directly after loading. If later add-ons are disabled by map/game loading code, this may mean that spurious registries by disabled add-ons are still in existence. In the worst case this means that add-ons are silently enabled they should not be.

As an example see this replay: [replays.zip](https://github.com/widelands/widelands/files/6575058/replays.zip)
It shows "No add-ons enabled", but fails to load because the Lapacho tree (introduced by the "Tropics" World add-on) is not registered.

To reproduce: 
- Install and enable the Tropics add-on. It's a World type add-on that adds several new trees.
- Start a new game on an official map.
- Since the enabled world add-ons are stored in the map file, you'd expect the game not to use the Tropics add-on even though it's enabled in the add-ons manager.
- Apparently, it is nonetheless possible for a Lapacho tree to be planted by a forester in this game.

The solution is to defer registering the descriptions until such a time when we know for sure that the list of add-ons that should be enabled in the game is in its final state. (It is initialized on creation with the currently enabled add-ons and modified later by random map generator, map loader, game loader and/or gameclient code). Essentially we'll need to refactor the load order for all game loading paths again. Related: #4786 #4834

Interestingly, this bug only seems to result in load failures when trying to watch a replay with a Lapacho in it. Loading an SP game with the impossible tree appears to work fine as this loading path apparently also registers the undesired add-on; in this case the bug is present both before and after loading so no problem (unless you remove or disable the Tropics add-on in the meantime, since it's not stored as a requirement for the savegame). Still need to check what this behaves like in MP savegames. – edit: On the host this fails with an unregistered object message when starting a new game. Loading a savegame without add-ons with Tropic enabled in the manager appears to be no problem. The replays are fine as well. So not critical.

Versions: Happens with both `1.0~git25322`[270dd13@master] (Release) and `1.0~git25323`[0f52e7f@editor-more-exception-safety] (Debug)

**Comment by Noordfrees** [comment]
SP savegame with supposedly no add-ons but with a Lapacho tree in it (can only be loaded if the Tropics add-on is enabled)
[DatGroteUtnaihenATL_lapacho.wgf.zip](https://github.com/widelands/widelands/files/6575625/DatGroteUtnaihenATL_lapacho.wgf.zip)
![grafik](https://user-images.githubusercontent.com/48293446/120311038-53864e00-c2d7-11eb-927d-44889b7f1a04.png)

**Comment by Noordfrees** [comment]
There's a very simple repro case, see https://www.widelands.org/forum/post/38995/

1. Enable the White Summer add-on
2. Start a new SP game on an official map
3. Build some foresters, speed up, wait a bit
4. White Summer trees get planted
5. Save and reload → `Unknown map object type 'linden_summer_old'`

Note that loading the savegame will fail even if the add-on remains enabled.

Retargeting as urgent.

**Comment by hessenfarmer** [comment]
ok tested it a bit. However I am not sure whether a specific behaviour is intended.  
I made a simple map with tropics and white summer enabled. This was correctly saved with the map. Then  I disabled the addons. When loading the map however the editor has both addons available. Only when I deinstall an addon the map is not loadable and I get the message that the addon is missing.   
Same for any savegame. as long as the addon is present on the machine. the game loads and works, whether the addon is enabled or not.
So if an addon is saved with a map or a game it is automatically enabled when loading such a file.
